src/pool-bin/libraries/BinPosition.sol

```diff
@@ -29,9 +30,13 @@ library BinPosition {
         // ref: https://github.com/Vectorized/solady/blob/main/src/tokens/ERC20.sol#L95
         // memory will be 12 bytes of zeros, the 20 bytes of address, 3 bytes for uint24
         assembly {
+            mstore(0x23, salt)
             mstore(0x03, binId)
             mstore(0x00, owner)
-            key := keccak256(0x0c, 0x17)
+            key := keccak256(0x0c, 0x37)
+            // only 0x00 ~ 0x40 is scratch space
```

Comment:
typo: `0x00 - 0x3f` is scratch space

Reply:
curious why is to `3f` not `40` ?

Reply:
because it starts from 0x00 😂  (2 slots i.e. 64 bytes in total)